packages/edit-navigation/src/store/test/utils.js

```diff
@@ -299,12 +341,269 @@ describe( 'computeCustomizedAttribute', () => {
+				url: '/example-category/',
+				description:
+					'Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+				rel: '',
+				className: '',
+				title: '',
+				id: 102,
+				type: 'category',
+				kind: 'taxonomy',
+				opensInNewTab: true,
 			},
+			clientId: 'navigation-link-block-client-id-3',
+			innerBlocks: [],
+			isValid: true,
+			name: 'core/navigation-link',
+		},
+		{
+			attributes: {
+				label: 'Example Tag',
+				url: '/example-tag/',
+				description: '',
+				rel: '',
+				className: '',
+				title: '',
+				id: 103,
+				type: 'tag',
+				kind: 'taxonomy',
+				opensInNewTab: false,
+			},
+			clientId: 'navigation-link-block-client-id-4',
+			innerBlocks: [],
+			isValid: true,
+			name: 'core/navigation-link',
+		},
+		{
+			attributes: {
+				label: 'Example Custom Link',
+				url: 'https://wordpress.org',
+				description: '',
+				rel: '',
+				className: '',
+				title: '',
+				type: 'custom',
+				kind: 'custom',
+				opensInNewTab: true,
+			},
+			clientId: 'navigation-link-block-client-id-5',
+			innerBlocks: [],
+			isValid: true,
+			name: 'core/navigation-link',
+		},
+	];
+
+	const menuItems = [
+		{
+			title: {
+				raw: 'Example Page',
+				rendered: 'Example Page',
+			},
+			url: '/example-page/',
+			description: 'Lorem ipsum dolor sit amet.',
+			xfn: [ 'friend', 'met' ],
+			classes: [ 'my-custom-class-one', 'my-custom-class-two' ],
+			attr_title: 'Example page link title attribute',
+			object_id: 100,
+			object: 'page',
+			type: 'post_type',
+			target: '_blank',
+		},
+		{
+			title: {
+				raw: 'Example Post',
+				rendered: 'Example Post',
+			},
+			url: '/example-post/',
+			description: 'Consectetur adipiscing elit.',
+			xfn: [ 'friend' ],
+			classes: [ 'my-custom-class-one' ],
+			attr_title: 'Example post link title attribute',
+			object_id: 101,
+			object: 'post',
+			type: 'post_type',
+			target: '',
+		},
+		{
+			title: {
+				raw: 'Example Category',
+				rendered: 'Example Category',
+			},
+			url: '/example-category/',
+			description:
+				'Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+			xfn: [ '' ],
+			classes: [ '' ],
+			attr_title: '',
+			object_id: 102,
+			object: 'category',
+			type: 'taxonomy',
+			target: '_blank',
+		},
+		{
+			title: {
+				raw: 'Example Tag',
+				rendered: 'Example Tag',
+			},
+			url: '/example-tag/',
+			description: '',
+			xfn: [ '' ],
+			classes: [ '' ],
+			attr_title: '',
+			object_id: 103,
+			object: 'tag',
+			type: 'taxonomy',
+			target: '',
+		},
+		{
+			title: {
+				raw: 'Example Custom Link',
+				rendered: 'Example Custom Link',
+			},
+			url: 'https://wordpress.org',
+			description: '',
+			xfn: [ '' ],
+			classes: [ '' ],
+			attr_title: '',
+			object: 'custom',
+			type: 'custom',
+			target: '_blank',
+		},
+	];
+
+	const blockAttrs = blocks.map( ( block ) => block.attributes );
+
+	describe( 'mapping block attributes to menu item fields', () => {
```

Comment:
Great idea @talldan. Thanks!

Reply:
I've decided to not include these types of tests because following [this discussion](https://github.com/WordPress/gutenberg/pull/31004#discussion_r623756348) we now no longer have lossless conversion in our code because we allow Core to set the defaults. Therefore a 1:1:1 conversion will no longer produce the results we expect.